### ISSUE TITLE
Server causes a system wide crash when moving LARGE videos to object storage

### ISSUE THREAD

**Issue by Kinuseka** [Chocobozzz/PeerTube#5458]
### Describe the current behavior

This has been unreported so I am not sure if it is actually just me, but I am running this server on a 2vcore 2gb of ram environment (already way above peertube's minimum). But when I move large videos usually starts at `800mb` then when peertube starts to move the video files to the object storage (occurs on any object storage `aws, b2, r2` ) the servers LITERALLY crashes.

I've investigated this after a while but I could never find error logs on peertube, other than checking `htop` and when the video finishes transcoding and starts moving the video, apparently my ram *SKYROCKETS* in usage. and then poof the entire system offline. Thus because of this behavior, this has forced me to NOT use higher quality video instead `480p, 360p` not because of performance but rather because of a memory dependent issue that has not been documented before.

# My Observations:

This is a weird behaviour, but this only occurs probably if the `filesize > memory` and during that process, it **LOADS** the entire file into RAM instead. I believe that should not be the case and shouldnt be loaded into memory. If better rather read the file in chunked and send those chunks instead less load in memory. 

Since this has not been reported before, I would like to also for other people that had encountered this issue to also post their experiences here if it had occured a similar experience.

### Steps to reproduce

1. Upload LARGE video ()
2. Wait for the video to transcode
3. Check result when it starts moving the video to object storage

### Describe the expected behavior

Uploaded video normally

### Additional information

* PeerTube instance:
  * URL: https://tube.kinu.m;
  * Version: 4.3.1
  * NodeJS version: v14.21.1
  * Ffmpeg version: 4.4.2

* Browser name, version and platforms on which you could reproduce the bug:
* Link to browser console log if relevant: OVH Cloud, EC2 instances and or any severs running under 2gb ram
* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`): No logs captured

Job is under failed tab
```
Job: 74 Type: move-to-object-storage Processed on 12/13/22, 11:07:53.349 PM Finished on 12/13/22, 11:09:00.966 PM
--
{   "videoUUID": "6bf4db35-96f5-427e-a1d2-8b77b18b33db",   "isNewVideo": true,   "previousVideoState": 2 }


```

**Comment by Chocobozzz** [comment]
Hi,

Can you display the `object_storage.max_upload_part` value of your `production.yaml`?

**Comment by Kinuseka** [comment]
> Hi,
> 
> Can you display the `object_storage.max_upload_part` value of your `production.yaml`?

```yaml
# Maximum amount to upload in one request to object storage
  max_upload_part: 5GB
 ```

**Comment by Chocobozzz** [comment]
Check configuration change when you upgrade: https://docs.joinpeertube.org/install-any-os?id=configuration

We reduced its value: https://github.com/Chocobozzz/PeerTube/blob/develop/config/production.yaml.example#L174

Try to change this setting and retry the upload

**Comment by Kinuseka** [comment]
@Chocobozzz 
Nope, It still does the same issue. I was able it to monitor the moving process, and yes the memory does get filled up like from 50% to 90% before the entire operating system completely went into memory overflow and crashed. bmon statistic show no packet was being transmitted during that process, which means that the process used for moving videos to object storage was trying to load it to memory first before sending it to object storage.

**Comment by Chocobozzz** [comment]
What if you change this value and put a lower one? Like 10MB.

**Comment by Kinuseka** [comment]
> What if you change this value and put a lower one? Like 10MB.

Okay transcoding took awhile, now I have realistic results,

Yes setting it to lower like 10mb certainly fixes it, I can now publish huge videos 1gb+ (My test was using 2gb video)
Cons is that it will cause more requests to your object storage (therefore might cost you ingress or API costs depending on the service used), the internet speed still depends on your Datacenter. Mine was at max 20mb/second so it took awhile.

**Comment by Chocobozzz** [comment]
AWS library we use to upload content to object storage seems to load chunks in memory. We have 4 concurrent uploads so you'll have to change this setting value according to your server memory/object storage provider limits. I don't think we can do anything else on PeerTube side.
